internal/cmd/backup.go

```diff
@@ -48,6 +54,52 @@ func createBackupFile(filename string) (*os.File, error) {
 	return f, nil
 }
 
+var (
+	missingAllowedTypes = regexp.MustCompile(`(\s*)(relation)(.+)(\/\* missing allowed types \*\/)(.*)`)
+	shortRelations      = regexp.MustCompile(`(\s*)relation [a-z][a-z0-9_]:(.+)`)
+)
+
+func filterSchemaDefs(schema, prefix string) (filteredSchema string, err error) {
+	// Remove any invalid relations generated from old, backwards-incompat
+	// Serverless permission systems.
+	schema = string(missingAllowedTypes.ReplaceAll([]byte(schema), []byte("\n/* deleted missing allowed type error */")))
+	schema = string(shortRelations.ReplaceAll([]byte(schema), []byte("\n/* deleted short relation name */")))
+
+	compiledSchema, err := compiler.Compile(compiler.InputSchema{Source: "schema", SchemaString: schema}, compiler.SkipValidation())
```

Comment:
let's not block on this since at least is a safe strategy. I think the scenario I reproduced with a test was precisely what the regexes above fix, so it may be difficult to actually make it err